lib/generators.nix

```diff
@@ -14,29 +14,84 @@
  * Documentation in the manual, #sec-generators
  */
 { lib }:
-with (lib).trivial;
-let
-  libStr = lib.strings;
-  libAttr = lib.attrsets;
-
-  inherit (lib) isFunction;
-in
 
-rec {
+let
+  inherit (builtins)
+    addErrorContext
+    attrNames
+    concatLists
+    concatStringsSep
+    elem
+    filter
+    head
+    isAttrs
+    isBool
+    isFloat
+    isInt
+    isList
+    isPath
+    isString
+    length
+    mapAttrs
+    match
+    replaceStrings
+    split
+    tail
+    toJSON
+    typeOf
+    ;
```

Comment:
Of these names, all but `match`, `split`, `toJSON`, and `typeof` could be imported from `lib` directly. Is there a preference?

Reply:
I'd prefer `lib` be used whenever possible. `lib` should ideally expose all `builtins`.

Reply:
Done.

Reply:
> lib should ideally expose all builtins.

Nice. I will abuse this from now on :)